From 593b9c07409cf732e83f1e80a60c2ecf3baac8ff Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Tue, 24 Dec 2024 11:08:26 -0800
Subject: [PATCH] Fix compile error on macOS

Error was:

  ../../common/Variant.hxx:52:65: error: no member named 'view' in 'std::ostringstream'
      Variant(Int32 i)  { buf().str(""); buf() << i; data = buf().view(); }
                                                            ~~~~~ ^
---
 src/common/Variant.hxx | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/common/Variant.hxx b/src/common/Variant.hxx
index 999762984..48c5c8831 100644
--- a/src/common/Variant.hxx
+++ b/src/common/Variant.hxx
@@ -49,13 +49,13 @@ class Variant
     Variant(string_view s) : data{s} { }
     Variant(const char* s) : data{s} { }
 
-    Variant(Int32 i)  { buf().str(""); buf() << i; data = buf().view(); }
-    Variant(uInt32 i) { buf().str(""); buf() << i; data = buf().view(); }
-    Variant(float f)  { buf().str(""); buf() << f; data = buf().view(); }
-    Variant(double d) { buf().str(""); buf() << d; data = buf().view(); }
-    Variant(bool b)   { buf().str(""); buf() << b; data = buf().view(); }
-    Variant(const Common::Size& s) { buf().str(""); buf() << s; data = buf().view(); }
-    Variant(const Common::Point& s) { buf().str(""); buf() << s; data = buf().view(); }
+    Variant(Int32 i)  { buf().str(""); buf() << i; data = buf().str(); }
+    Variant(uInt32 i) { buf().str(""); buf() << i; data = buf().str(); }
+    Variant(float f)  { buf().str(""); buf() << f; data = buf().str(); }
+    Variant(double d) { buf().str(""); buf() << d; data = buf().str(); }
+    Variant(bool b)   { buf().str(""); buf() << b; data = buf().str(); }
+    Variant(const Common::Size& s) { buf().str(""); buf() << s; data = buf().str(); }
+    Variant(const Common::Point& s) { buf().str(""); buf() << s; data = buf().str(); }
 
     // Conversion methods
     const string& toString() const { return data; }
-- 
2.47.1

